File: js/src/__tests__/RuleImpl.js
import Rule from "../model/Rule";
import {RuleBuilder} from "../builder";

class RuleImpl extends Rule{
    title;
    description;
    constructor(condition,promotion,title,description) {
        super(condition,promotion);
        this.title = title;
        this.description = description;
    }
}

class RuleImplBuilder extends RuleBuilder{
    _title;
    _description;
    constructor() {
        super(null);
    }

    title(t){
        this._title = t;
        return this;
    }

    description(d){
        this._description = d;
        return this;
    }

    build(){
        let r = super.build();
        r.title = this._title;
        r.description = this._description;
        return r;
    }
}

test('-',()=>{});

export { RuleImpl, RuleImplBuilder };

File: java/src/test/java/com/github/thinhunan/wonder8/promotion/rule/model/RuleImpl.java
package com.github.thinhunan.wonder8.promotion.rule.model;


public class RuleImpl extends Rule {

    public static RuleImplBuilder myBuilder(){
        return new RuleImplBuilder();
    }

    String description;

    String title;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}
